docs/topics/conditional-view-processing.txt

```diff
@@ -198,6 +200,12 @@ to compute the ETag and last modification values in all situations. In fact,
 you **should** use the same functions, so that the same values are returned
 every time.
 
+.. note::
+
+    Django does not automatically set the ``ETag`` and ``Last-Modified``
+    headers for non-safe methods. If you wish to return them they should be
+    set by your view function.
```

Comment:
I think it would be helpful to include a brief warning and a link to the RFC here, something like: "If you wish to return them they should be set by your view function, though be sure to note the caveats in [https://tools.ietf.org/html/rfc7231#section-4.3.4] about doing so in response to `PUT` methods." (IIRC we have a Sphinx RFC tag.)